### PR TITLE
Implemented logic to fetch products and render the data on selectProduct page(#28zu62r)

### LINKED ISSUE #28: Show product information on the product select page.
On the product select page:

User should be able to search products like our other apps.
Results should be grouped by parent product
Products should be selectable

### PR DESCRIPTION
**Pull request by disha1202**
Closes #28 
Screenshot from 2022-04-27 18-21-51

### CODE REVIEW

File: src/views/SelectProduct.vue [hunk 1]
```diff
@@ -47,37 +48,29 @@
...
-            <ion-card>
-              <Image src="https://cdn.shopify.com/s/files/1/0069/7384/9727/products/test-track.jpg?v=1626255137" />
-              <ion-item lines="none">
-                <ion-label>
-                  SKU
-                  <p>Color: Blue</p>
-                  <p>Size: XL</p>
-                </ion-label>
-                <ion-checkbox slot="end" />
+                <ion-checkbox v-model="variant.isSelected" @click="select(variant)" slot="end" />
```
Comment (ymaheshwari1) [inline src/views/SelectProduct.vue:66]: `select` method is not defined.

File: src/store/modules/product/mutations.ts [hunk 2]
```diff
@@ -6,6 +6,13 @@ const mutations: MutationTree <ProductState> = {
   [types.PRODUCT_SEARCH_UPDATED] (state, payload) {
     state.products.list = payload.products;
     state.products.total = payload.totalProductsCount;
+  },
+  [types.PRODUCT_CACHED_UPDATED](state, payload) {
+    state.cached = payload.cached
+  },
+  [types.PRODUCT_LIST_UPDATED] (state, payload) {
```
Comment (ymaheshwari1) [inline src/store/modules/product/mutations.ts:13]: I think we can use the same mutation `PRODUCT_SEARCH_UPDATED` or rename that mutation if it's not used anywhere else.

File: src/store/modules/product/actions.ts [hunk 3]
```diff
@@ -46,6 +46,98 @@ const actions: ActionTree<ProductState, RootState> = {
...
+            isSelected: false,
+            brandName: virtual?.brandName,
+            productName: virtual?.productName,
+            internalName: virtual?.internalName,
+            mainImageUrl: virtual?.mainImageUrl,
+            featureHierarchy: virtual?.featureHierarchy
+          }
+        })
+
+        // We are commenting this code because we will be releasing this feature in next release.
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:123]: Remove this unwanted comment.

File: src/store/modules/product/actions.ts [hunk 4]
```diff
@@ -46,6 +46,98 @@ const actions: ActionTree<ProductState, RootState> = {
...
+
+        let productIds: any = new Set();
+        products.forEach((product: any) => {
+          if(product.productId) productIds.add(product.productId);
+        })
+        productIds = [...productIds]
+        const productInformation = await this.dispatch("product/fetchProducts", { productIds });
+
+        products = products.map((product: any) => {
+          product.variants.forEach((variant: any) => {
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:107]: Use map instead

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/SelectProduct.vue (hunk 1 above)
Comment (disha1202) [reply src/views/SelectProduct.vue]: Removed.

File: src/store/modules/product/mutations.ts (hunk 2 above)
Comment (disha1202) [reply src/store/modules/product/mutations.ts:9]: Removed PRODUCT_SEARCH_UPDATED mutation.

File: src/store/modules/product/actions.ts (hunk 3 above)
Comment (disha1202) [reply src/store/modules/product/actions.ts]: Removed unwanted code.

File: src/store/modules/product/actions.ts (hunk 4 above)
Comment (disha1202) [reply src/store/modules/product/actions.ts]: Used map instead of for each.

**Comment by ymaheshwari1** [comment]
Closing as not needed.

**Pull request closed without merging**